resources/js/tableofcontent.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Construct recurseData.toc as a JSON-like that contains each header's text
+ * anchor, and their list of child headers.
+ * 
+ * References:
+ * https://stackoverflow.com/questions/187619/is-there-a-javascript-solution-to-generating-a-table-of-contents-for-a-page
+ * @param elem DOM element
+ * @param recurseData Object with fields: { toc : Object, count : int, parent : Object }
+ */
+function recurseChild(elem, recurseData)
+{
+    if (elem !== "undefined") 
```

Comment:
1. IMHO, `undefined` shouldn't be quoted
2. Is this check needed here at all? I believe it has to be performed only before the top-level call of `recureChild()` (to guard against the missing `<body>`).

Reply:
Done